src/pyspiffe/config.py

```diff
@@ -0,0 +1,21 @@
+import os
+from typing import Dict
+
+SPIFFE_ENDPOINT_SOCKET = 'SPIFFE_ENDPOINT_SOCKET'
+
+
+class ConfigSetter:
+    def __init__(self):
+        self._apply_default_config()
+        self._apply_environment_variables()
+
+    def get_config(self) -> Dict:
+        return self._config
+
+    def _apply_default_config(self):
+        self._config = {
+            SPIFFE_ENDPOINT_SOCKET: None,
+        }
+
+    def _apply_environment_variables(self):
+        self._config[SPIFFE_ENDPOINT_SOCKET] = os.environ.get(SPIFFE_ENDPOINT_SOCKET)
```

Comment:
I suggest the following is also checked: 1) environment variable is set;  2) value follow expected format "unix://...." or "tcp://..." and 3) The value set in the environment variable respects maximum permitted length of a path (UNIX_PATH_MAX) in linux.

I wonder if we should also check for the other validations mentioned by SPIFFE specification e.g: if unix scheme, authority component MUST NOT be checked, only scheme and paths are present... and for tcp, host authority MUST be set to an IP address, and make sure scheme, host and port components are set.

If the decision is to follow the path for the extra validation scenario, I would suggest creating a SpiffeEndpointSocket to hold all of these and ConfigSetter could make usage of it.

Reply:
I agree with @glaucimar, we should validate the address following the spiffe spec: https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE_Workload_Endpoint.md#4-locating-the-endpoint
Also it's a good idea moving the validations to another class. In the go-spiffe and java-spiffe there are tests enforcing those constraints.

Reply:
thanks folks, it makes sense to add these validations... the only thing I'm not sure is if we need to check if the env var is set... the SPIFFE spec says `Clients may be explicitly configured with the socket location, or may utilize the well-known environment variable SPIFFE_ENDPOINT_SOCKET. If not explicitly configured, conforming clients MUST fall back to the environment variable.` what makes me think that the client can be explicitly instantiated with a given socket, and the env var is not always mandatory

Reply:
Seems reasonable to enforce a path per the SPIFFE specification.